lighthouse-core/audits/treemap-data.js

```diff
@@ -0,0 +1,315 @@
+/**
+ * @license Copyright 2020 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview
+ * Creates treemap data for treemap app.
+ */
+
+const Audit = require('./audit.js');
+const JsBundles = require('../computed/js-bundles.js');
+const UnusedJavaScriptSummary = require('../computed/unused-javascript-summary.js');
+const ModuleDuplication = require('../computed/module-duplication.js');
+const NetworkRecords = require('../computed/network-records.js');
+const ResourceSummary = require('../computed/resource-summary.js');
+
+/**
+ * A collection of root nodes, grouped by type.
+ * @typedef {Record<string, RootNode[]>} TreemapData
+ */
+
+/**
+ * Ex: https://gist.github.com/connorjclark/0ef1099ae994c075e36d65fecb4d26a7
+ * @typedef RootNode
+ * @property {string} name Arbitrary name identifier. Usually a script url.
+ * @property {Node} node
+ */
+
+/**
+ * @typedef Node
+ * @property {string} name Arbitrary name identifier. Usually a path component from a source map.
+ * @property {number} resourceBytes
+ * @property {number=} unusedBytes
+ * @property {string=} duplicate If present, this module is a duplicate. String is normalized source path. See ModuleDuplication.normalizeSource
```

Comment:
it would be great to rename this to something more descriptive (e.g. it sounds like a boolean), even if it's something terrible like `duplicateCanonicalName` or whatever